backend/app/key_migration.py

```
@@ -0,0 +1,418 @@
+        """)
+        expected_field_ids = {row[0] for row in cursor.fetchall()}
+
+        # Check for duplicates in request
+        request_user_ids = [u.id for u in request.users]
+        request_field_ids = [f.id for f in request.field_values]
+
+        if len(request_user_ids) != len(set(request_user_ids)):
+            raise HTTPException(
+                status_code=400,
+                detail="Duplicate user IDs in request. Each user must appear exactly once."
+            )
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)